spec/models/story_spec.rb

```diff
@@ -7,6 +7,34 @@
     expect(s.short_id).to match(/^\A[a-zA-Z0-9]{1,10}\z/)
   end
 
+  it "is not valid without a short id" do
+    s = create(:story, :title => "goodbye", :url => "http://example.com/")
+    s.short_id = nil
+
+    expect(s).to_not be_valid
```

Comment:
Please call `.valid?` and then assert that the specific field is invalid rather than the whole model. This prevents false positive tests where this validation fails but something unrelated about the model is invalid and the test continues passing.

Reply:
Is this format ok?
```
expect(s.valid?).to be false
expect(s.errors[:short_id]).not_to be_empty
```

Reply:
RSpec aslo has sexy syntax, so you could say something like this:
```ruby
expect(s).not_to be_valid
expect(s.errors[:sort_id]).to include(expected_error)
```
The above assumes that `expected_error` is already defined above (such as in a let-block).

Reply:
You can skip `expect(s.valid?).to be false`, it's redundant with testing that there's an error on the next field.

I agree with @jaysonvirissimo, might as well check for the particular error. There's a pattern of checking error message in #763 you can reuse.